### PR TITLE
Add encoder: Decimal to Decimal128

### PR DESCRIPTION
**Pull request by dhensen**
Add an encoder to encode `decimal.Decimal` instances into `from bson.decimal128.Decimal128`.
Mongo stores this as NumberDecimal:
image

### CODE REVIEW

**Comment by dhensen** [comment]
CI is failing, I think this will be fixed by #32

File: tests/encoders/mongodb.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 """Tests for mongodb encoders"""
 import pytest
+from bson.decimal128 import Decimal128
+from decimal import Decimal
```
Comment (i8enn) [inline tests/encoders/mongodb.py:4]: What for, if not used?

File: pydantic_odm/encoders/mongodb.py [hunk 2]
```diff
@@ -42,20 +40,52 @@ def _convert_enums(
     # Iterate passed data
     for key, value in iterator:
         # Replace enum object to enum value
```
Comment (i8enn) [inline pydantic_odm/encoders/mongodb.py]: Replace comment to:
```python
# Convert and replace value with transform function
```

File: pydantic_odm/encoders/mongodb.py [hunk 3]
```diff
@@ -18,14 +20,10 @@ def __call__(self, data: "DictStrAny") -> "DictStrAny":
         raise NotImplementedError()
 
 
-def _convert_enums(
-    data: Union["DictStrAny", List[Any]]
+def _recursive_iterator(
+    data: Union["DictStrAny", List[Any]], transform_func
```
Comment (i8enn) [inline pydantic_odm/encoders/mongodb.py]: You need to create a type for the transform functions and specify here (for mypy)

**Review by i8enn — CHANGES_REQUESTED** [review]
Thanks! You are making a good contribution! ;)

I did a review, and for your PR to be accepted, I need you to write tests and fix code-style bugs (flake8, mypy and etc.)

File: tests/encoders/mongodb.py (hunk 1 above)
Comment (dhensen) [reply tests/encoders/mongodb.py:4]: Oops, I stashed tests and forgot to unstash. No idea why imports were commited though.. Fixed in 7c1449cf094951c2e4088178ae17bf96dc9f6db8

File: pydantic_odm/encoders/mongodb.py (hunk 3 above)
Comment (dhensen) [reply pydantic_odm/encoders/mongodb.py]: Typings added in: dada6db94ba58c708033c7180b5e58023323284e

Hmm, I am having issues with flake8 conflicting with what black does. When I run `make lint` I get:

```
$ make lint
flake8 pydantic_odm/ tests/
tests/encoders/mongodb.py:95:52: E231 missing whitespace after ','
tests/encoders/mongodb.py:96:55: E231 missing whitespace after ','
tests/encoders/mongodb.py:100:62: E231 missing whitespace after ','
tests/encoders/mongodb.py:101:65: E231 missing whitespace after ','
tests/encoders/mongodb.py:107:72: E231 missing whitespace after ','
tests/encoders/mongodb.py:111:75: E231 missing whitespace after ','
make: *** [lint] Error 1
```
flake8 immediately fails over some whitespace that black does not fix for me (`make format` should fix that right?).
Then ofcourse the mypy, isort and black checks are not executed. Do you have this issue as well? If not, maybe one of us has some local configuration hanging somewhere.

File: pydantic_odm/encoders/mongodb.py (hunk 2 above)
Comment (dhensen) [reply pydantic_odm/encoders/mongodb.py]: Fixed in dada6db94ba58c708033c7180b5e58023323284e

File: pydantic_odm/encoders/mongodb.py (hunk 3 above)
Comment (dhensen) [reply pydantic_odm/encoders/mongodb.py]: Googled and found https://github.com/psf/black/issues/1289 which points to https://github.com/psf/black/issues/1288 for the fix. Basically this is some kind of magic comma happening here. Apparantly if I remove it manually, black wont make it come back.

Comment (dhensen) [reply pydantic_odm/encoders/mongodb.py]: Resolved linting issues with: 135fb9cce1fe129ae7974d155e2a099e671a10f0

**Review by i8enn — APPROVED** [review]

**Comment by i8enn** [comment]
@dhensen well done! Thanks! I'm merge this PR.

**Pull request merged**